contract/models/contract.py

```diff
@@ -217,12 +212,61 @@ def action_contract_send(self):
             'context': ctx,
         }
 
+    @api.model
+    def _get_recurring_create_invoice_domain(self, contract=False):
+        domain = []
+        date_ref = fields.Date.context_today(self)
+        if contract:
+            contract.ensure_one()
+            date_ref = contract.recurring_next_date
+            domain.append(('id', '=', contract.id))
+        domain.extend(
+            [
+                ('recurring_invoices', '=', True),
+                ('recurring_next_date', '<=', date_ref),
+            ]
+        )
+        return domain
+
+    @api.multi
+    def _get_lines_to_invoice(self, date_ref=False):
+        self.ensure_one()
+        if not date_ref:
+            date_ref = fields.Date.context_today(self)
+        return self.recurring_invoice_line_ids.filtered(
+            lambda l: not l.is_canceled and l.recurring_next_date <= date_ref)
+
     @api.multi
     def recurring_create_invoice(self):
-        return self.env[
-            'account.analytic.invoice.line'
-        ].recurring_create_invoice(self)
+        invoice_model = self.env['account.invoice']
+        invoices_values = []
+        for contract in self:
+            contract_lines = contract._get_lines_to_invoice()
+            if not contract_lines:
+                continue
+            invoice_values = contract._prepare_invoice(
+                contract.recurring_next_date)
+            for line in contract_lines:
+                invoice_values.setdefault('invoice_line_ids', [])
+                invoice_values['invoice_line_ids'].append(
+                    (0, 0, line._prepare_invoice_line(False))
```

Comment:
Here a keyword argument is really useful so the poor reader does not need to ask himself what False means.